### PR TITLE
ODH-1813: Updated logic of BA job.

### PR DESCRIPTION
**Pull request by ltv12**

### CODE REVIEW

File: src/test/it/core/base_spark_test_case.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def _test_pipeline(self, configuration_path, processor_creator, input_dir, expec
     def __read_expected_result(expected_result_file):
         with open(expected_result_file) as json_file:
             lines = filter(
-                lambda json_line: json_line.strip() != "",
+                lambda json_line: (json_line.strip() != "") & ~(json_line.strip().startswith("//")),
```
Comment (chermenin) [inline src/test/it/core/base_spark_test_case.py]: why not `json_line.strip() != "" and not json_line.strip().startswith("//")` ?

File: src/applications/basic_analytics/prodis/ws_driver.py [hunk 2]
```diff
@@ -16,46 +16,43 @@ class ProdisWS(BasicAnalyticsProcessor):
...
-                        when(
-                            (col("level") == "INFO") &
-                            (
-                                (col("message").contains("successfully initialized port"))
-                                |
-                                ~(col("message").rlike("^[^']*(['][^']*?['])*[^']*succe.*"))
-                            ),
-                            "SUCCESS")
+                        when((col("level") == "INFO")
+                             &
```
Comment (chermenin) [inline src/applications/basic_analytics/prodis/ws_driver.py]: can we use `and` here? and it's really needed a separate line for that?

Comment (ltv12) [reply src/applications/basic_analytics/prodis/ws_driver.py]: No we can't, catalyst implicitly use this symbols

Comment (chermenin) [reply src/applications/basic_analytics/prodis/ws_driver.py]: ok

File: src/applications/basic_analytics/prodis/ws_driver.py [hunk 3]
```diff
@@ -16,46 +16,50 @@ class ProdisWS(BasicAnalyticsProcessor):
...
-        return [counts]
+            .withColumn("level", col("level_updated"))
+
+        stream_with_mapped_groups = filtered_stream \
+            .withColumn("group", custom_translate_like(source_field=lower(col("thread_name")),
+                                                       mappings_pair=self.__mapping_thread_name_to_group,
+                                                       default_value="unclassified"))
+
+        count_by_level = stream_with_mapped_groups \
+            .where(~((col("level") == "SUCCESS") & ~(lower(col("message")).like("%succe%")))) \
```
Comment (vunder-kind) [inline src/applications/basic_analytics/prodis/ws_driver.py]: `.where((col("level") != "SUCCESS") | lower(col("message")).like("%succe%"))) \`

**Pull request merged**